documentation/source/proposals/dep-0013-multi-assignment.rst

```diff
@@ -0,0 +1,236 @@
+last *place* may be preceded by ``#rest``, in which case it is assigned a sequence
+containing all the remaining values.
+
+If there are more *places* than values returned by ``new-values``, the extra *places* are
+assigned the value :drm:`#f`.
+
+The name ``_`` (underscore) may be used to indicate that a value is to be ignored. For
+example, ``(_, remainder) := round(3.2)`` will ignore the first return value and assign
+``remainder`` the value ``0.2``.
+
+The order of assignment of values to *places* is unspecified.
+
+
+BNF Changes
+-----------
+
+No changes are required to the Dylan BNF as described in the DRM.  ``LEAF := EXPRESSION``
+is already valid and ``( EXPRESSION )`` syntax is a valid ``LEAF`` production.
+
+
+Rationale
+=========
+
+Multiple value assignment syntax naturally matches the existing multiple value binding
+syntax provided by :drm:`let` and can occasionally be a useful tool to make code more
+concise.
+
+
+Examples
+========
+
+Some examples from the Open Dylan sources that can be made more concise:
+
+.. code:: dylan
+
+   // This occurs a lot in dfmc/conversion/convert.dylan
+   let (f, l) = join-2x1!(first, last, comp); first := f;  last := l; // old
+   (first, last) := join-2x1!(first, last, comp);                     // new
+
+.. code:: dylan
+
+   // old:
+   define sealed method set-union!
+       (set1 :: <bit-set>, set2 :: <bit-set>) => (set1 :: <bit-set>)
+     let (vector, pad)
+       = bit-vector-or!(set1.member-vector, set2.member-vector,
+                        pad1: set1.member-vector-pad,
+                        pad2: set2.member-vector-pad);
+     set1.member-vector := vector;
+     set1.member-vector-pad := pad;
+     set1
+   end method;
+
+   // new:
+   define sealed method set-union!
+       (set1 :: <bit-set>, set2 :: <bit-set>) => (set1 :: <bit-set>)
+     (set1.member-vector, set1.member-vector-pad)
+       := bit-vector-or!(set1.member-vector, set2.member-vector,
+                         pad1: set1.member-vector-pad,
+                         pad2: set2.member-vector-pad);
+     set1
+   end method;
+
+.. code:: dylan
+
+   // old:
+   for (i from 0 below n)
+     let (thread, result) = join-thread(threads[i]);
+     results[i] := result;
+   end for;
+
+   // new:
+   for (i from 0 below n)
+     (_, results[i]) := join-thread(threads[i]);
+   end for;
+
+.. code:: dylan
+
+   // old:
+   define method stop-profiling-type
+       (state :: <profiling-state>, keyword :: <cpu-profiling-type>) => ()
+     when (element(state, #"cpu-profiling", default: #f))
+       let (seconds, microseconds) = timer-stop(state[#"cpu-profiling-timer"]);
+       state[#"cpu-time-seconds"]      := seconds;
+       state[#"cpu-time-microseconds"] := microseconds;
+       state[#"cpu-profiling"]         := #f;
+       state[#"cpu-profiling-timer"]   := #f;
+     end
+   end method stop-profiling-type;
+
+   // new:
+   define method stop-profiling-type
```

Comment:
Maybe this can be an example of more places than values:

```
define method stop-profiling-type
       (state :: <profiling-state>, keyword :: <cpu-profiling-type>) => ()
     when (element(state, #"cpu-profiling", default: #f))
      (state[#"cpu-time-seconds"],
       state[#"cpu-time-microseconds"],
       state[#"cpu-profiling"],
       state[#"cpu-profiling-timer"])
         := timer-stop(state[#"cpu-profiling-timer"]);
     end
 end method stop-profiling-type;
```

Reply:
I would personally avoid doing that because if `timer-stop` is changed such that it returns *more* values (a change that is normally safe to make), this code is likely to break.